src/voip_patrol/voip_patrol.cc

```diff
@@ -454,6 +454,12 @@ void TestCall::onCallState(OnCallStateParam &prm) {
 		test->call_id = getId();
 		test->sip_call_id = ci.callIdString;
 	}
+	if (test && (ci.lastStatusCode == 180 || ci.lastStatusCode == 183)) {
+		if (test->ring_duration > 0) {
+			LOG(logINFO) <<__FUNCTION__<<": Waiting "<<test->ring_duration <<" seconds ring_duration";
+			pj_thread_sleep(test->ring_duration * 1000);
```

Comment:
Just to mention, that string is a blocking operation for all `voip_partol` execution and will block any other operations, that should be executed within `ring_duration` time.

Reply:
I see, wondering if that is an issue tho during requested `ring_duration` time?  
does it block incoming CANCEL requests from UAC for example?

Reply:
Yes it is the case, I'll rethink this.. thanks you!